draft-ietf-webtrans-overview.md

```diff
@@ -260,7 +298,67 @@ Streams SHOULD be sufficiently lightweight that they can be used as messages.
 Data sent on a stream is flow controlled by the transport protocol.  In addition
 to flow controlling stream data, the creation of new streams is flow controlled
 as well: an endpoint may only open a limited number of streams until the peer
-explicitly allows creating more streams.
+explicitly allows creating more streams.  From the perspective of the client,
+this is presented as a size-bounded queue of incoming streams.
+
+Any WebTransport protocol SHALL provide the following operations on the
+session:
+
+create a unidirectional stream
+: Creates an outgoing unidirectional stream; this operation may block until the
+  flow control allows for it to be completed.
+
+create a bidirectional stream
+: Creates an outgoing bidirectional stream; this operation may block until the
+  flow control allows for it to be completed.
+
+receieve a unidirectional stream
+: Removes a stream from the queue of incoming unidirectional streams, if one is
+  available.
+
+receive a bidirectional stream
+: Removes a stream from the queue of incoming unidirectional streams, if one is
+  available.
+
+Any WebTransport protocol SHALL provide the following operations on an
+individual stream:
+
+send bytes
+: Add bytes into the stream send buffer.  The sender can also indicate a FIN,
+  signalling the fact that no new data will be send on the stream.  Not
+  applicable for incoming unidirectional streams.
+
+receive bytes
+: Removes bytes from the stream receive buffer.  FIN can be received together
+  with the stream data.  Not applicable for outgoing unidirectional streams.
+
+abort send side
+: Sends a signal to the peer that the write side of the stream has been aborted.
+  Discards the send buffer; if possible, no currently outstanding data is
+  transmitted or retransmitted.  An unsigned 8-bit error code can be supplied
+  as a part of the signal to the peer.
+
+abort receive side
+: Sends a signal to the peer that the read side of the stream has been aborted.
+  Discards the receive buffer; the peer is typically expected to abort the
+  corresponding send side in response.  An unsigned 8-bit error code can be
+  supplied as a part of the signal to the peer.
+
+Any WebTransport protocol SHALL provide the following events for an individual
+stream:
+
+send side aborted
+: Indicates that the peer has aborted the corresponding receive side of the
+  stream.  An unsigned 8-bit error code from the peer may be available.
+
+receive side aborted
+: Indicates that the peer has aborted the corresponding send side of the
+  stream.  An unsigned 8-bit error code from the peer may be available.
+
+Data Recvd state reached
+: Indicates that no further data will be transmitted or retransmitted, and that
```

Comment:
FIN needs to be ACKed too, right?

Reply:
"no further data will be transmitted or retransmitted" should imply that (since FIN will be retransmitted).  I don't want to write down ACK specifically, since in H/2, we have no way of knowing when things get ACKed.